validate/validator.py

```diff
@@ -22,7 +23,14 @@
 }
 
 def validate(proto_message):
-    func = file_template(proto_message)
+    klass = type(proto_message)
+    klass.__hash__ = lambda self: str(klass).__hash__()
+    return validate_inner(klass)
+
+# Cache generated functions by class name.
+@lru_cache()
+def validate_inner(hashable):
+    func = file_template(hashable)
```

Comment:
Can we leave the argument name as `proto_message` or maybe `proto_class` instead of changing it to `hashable`? The fact that it needs to be hashable isn't actually relevant to this definition, just for the annotation and at the call site.